### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by kartikmohta**
- Allow camera fps to be set
- Use image_transport to publish images

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -74,7 +74,6 @@ if (camera_LIBRARIES)
         ${catkin_LIBRARIES}
         ${OpenCV_LIBS}
         snap_cam
-        klt_feature_tracker
```
Comment (ChristophTobler) [inline CMakeLists.txt:77]: This is needed for pure cmake build

**Comment by kartikmohta** [comment]
Fixed the pure cmake build, please take a look.

**Comment by ndepal** [comment]
What's the advantage of putting `klt_feature_tracker` into a variable? The `CMakeLists.txt` seems to just get longer without any actual change.

**Comment by kartikmohta** [comment]
It fails to build using `catkin_tools` when the library name is explicitly passed in (see the error below). When building with `catkin`, the correct library name would be passed through the `${catkin_LIBRARIES}` variable anyway, so no need to add `klt_feature_tracker` separately.

```
$ catkin build snap_cam 
--------------------------------------------------------------
Profile:                     default
Extending:          [cached] /opt/ros/jade
Workspace:                   /home/linaro/programs/ros
--------------------------------------------------------------
Source Space:       [exists] /home/linaro/programs/ros/src
Log Space:          [exists] /home/linaro/programs/ros/logs
Build Space:        [exists] /home/linaro/programs/ros/build
Devel Space:        [exists] /home/linaro/programs/ros/devel
Install Space:      [unused] /home/linaro/programs/ros/install
DESTDIR:            [unused] None
--------------------------------------------------------------
Devel Space Layout:          merged
Install Space Layout:        None
--------------------------------------------------------------
Additional CMake Args:       -DCMAKE_BUILD_TYPE=Release
Additional Make Args:        None
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
--------------------------------------------------------------
Whitelisted Packages:        None
Blacklisted Packages:        None
--------------------------------------------------------------
Workspace configuration appears valid.
--------------------------------------------------------------
[build] Found '7' packages in 0.0 seconds.                                                                                                 
[build] Package table is up to date.                                                                                                       
Starting  >>> klt_feature_tracker                                                                                                          
Finished  <<< klt_feature_tracker                [ 0.6 seconds ]                                                                           
Starting  >>> snap_cam                                                                                                                     
___________________________________________________________________________________________________________________________________________
Errors     << snap_cam:make /home/linaro/programs/ros/logs/snap_cam/build.make.003.log                                                     
/usr/bin/ld: cannot find -lklt_feature_tracker
collect2: error: ld returned 1 exit status
make[2]: *** [/home/linaro/programs/ros/devel/lib/snap_cam/optical_flow] Error 1
make[1]: *** [CMakeFiles/optical_flow.dir/all] Error 2
make: *** [all] Error 2
cd /home/linaro/programs/ros/build/snap_cam; catkin build --get-env snap_cam | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -
...........................................................................................................................................
Failed     << snap_cam:make                      [ Exited with code 2 ]                                                                    
Failed    <<< snap_cam                           [ 1.0 seconds ]
```

**Comment by ndepal** [comment]
good to know. We are building with `catkin_make`, which builds just fine.

**Pull request merged**